Formula/g/gstreamer.rb

```diff
@@ -102,13 +104,13 @@ class Gstreamer < Formula
   end
 
   def python3
-    which("python3.11")
```

Comment:
why this change?

Reply:
https://github.com/Homebrew/homebrew-core/pull/142566#issuecomment-1784237103

Reply:
The current problem is `system python3, "-c"` failing because Python 3.12 is not the default (i.e. `python3` does not exist). Changing it to `system "python3.12", "-c"` should fix it, but the CI-long-timeout tag is currently occupied.

Reply:
noted, let me update the build for it

Reply:
actually no need to change to "python3.12", `which("python3.12")` should just work for gstreamer.